veps/sig-compute/secure-execution.md

```diff
@@ -0,0 +1,60 @@
+# Overview
+Implement support for IBM Z Secure Execution VMs in kubevirt
+
+## Motivation
+To enable customers to deploy secure workloads on their IBM Z machines using kubevirt.
+
+## Goals
+To enable kubevirt to run Secure Execution VMs on s390x
+
+## Non Goals
+Changing how existing VMs run.
+
+## Definition of Users
+- IBM Z users wanting to secure their new VMs using Secure Execution
+- IBM Z users wanting to migrate existing Secure Execution VMs to kubevirt
+
+## User Stories
+(list of user stories this design aims to solve)
+
+## Repos
+https://github.com/kubevirt/kubevirt
+https://github.com/kubevirt/api
+
+# Design
+For implementing this feature, the api would need to be expanded to have a Secure Execution option under Launch Security next to AMD SEV.
+Similiar to SEV it would be first behind a Feature Gate.
+
+Afterwards kubevirt would need to be expanded to handle setting the correct Launch Security type for Secure Execution.
+To achieve this, the virt-launcher pod would currently need to be run as privileged for Secure Execution VMs, as Libvirt checks for `/sys/firmware/uv` when verifying that the Host supports Secure Execution VMs. Unfortunately `/sys/firmware` is intentionally empty in unprivileged pods and kubernetes prevents mounting anything under this path. There are potentially Feature Gates in kubelet to allow this, however i have not yet been able to get this to work. From what i gathered you would need to at least enable `ProcMountType` and `UserNamespacesSupport` to be able to launch a Secure Execution VM inside of an unprivileged pod.
+To avoid using privileged pods, the first implementation should explicitly not add launchSecurity to the xml file, but instead do the needed changes in the xml directly. This ensures that libvirt does not check if Secure Execution is supported, enabling virt-launcher to run unprivileged. The drawback of this is of course, that kubevirt needs to ensure all devices use iommu, otherwise they will not be visible to the VM, leading to potentially hard to debug, unkown behaviour and errors.
```

Comment:
Is this using sidecars for changing the XML?
In any case, it sounds like a bug in libvirt. If libvirt is running in session mode, it should assume that the environment has already been prepared

Reply:
It is not using sidecars for changing the xml, as what is needed is setting `iommu=on` for all devices. This is already done in kubevirt when using SEV, so the code is already mostly in place.

Reply:
I have simplified this passage and dropped mentions of launch security type and privileged pods as they are not needed.